core/controllers/base.py

```diff
@@ -264,10 +267,10 @@ def dispatch(self):
             return
 
         try:
-            if 'csrf' in self.request.uri:
-                if self.gae_id and not self.user_id:
-                    raise self.UnauthorizedUserException(
-                        'User details not found.')
+            # If this is a CSRF request and the user is not yet loaded produce
+            # an error.
```

Comment:
Can you please add a sentence or two here explaining more clearly the situations in which this might legitimately occur? "The user is not yet loaded" sound weird and seems like unexpected behaviour, so explaining all the stuff about datastore consistency etc. here (with links) would be helpful.

Reply:
Done.